docs/spec/intermediate_representation.md

```diff
@@ -476,6 +476,27 @@ for details on the field definition format.
 
 Note, the difference between safe and unsafe arguments are explained in the docs of [conjure-java-runtime](https://github.com/palantir/conjure-java-runtime#error-propagation).
 
+## Extensions
+
+The "extensions" section is a map of named extensions. The structure of each extension is user defined and
+any consumer of an extension should treat its presence as optional and be tolerant of unknown fields. Extensions must
+not modify the semantics of definitions and are intended only for conveying metadata about the definition.
+
+Example extension definition:
+```json
+{
+  "extensions": {
+    "product-dependencies": [
+      {
+        "minimum-version": "1.0.0",
+        "recommended-version": "1.1.0",
+        "maximum-version": "1.x.x"
```

Comment:
the dashes are not conjure expressible - could we go camel?

Reply:
I was going for consistency with what is included in manifests (JAR, and SLS)

Reply:
I think i'd prefer conjure to be internally consistent, and accept that this makes us inconsistent with the manifests out there in the wild... at least with jackson we can easily deserialize both but serialize as the dashes so you don't need two classes?

Reply:
I'd prefer to present a different example in the spec than change the structure of the product dependencies in production

Reply:
I'm just saying we've worked so hard to try and make conjure this nice clean world where things are all consistent, and I don't want to compromise that over the camel vs kebab thing. Not saying we need to change what's used in production, I was thinking that we can just do a translation?